x-pack/plugins/fleet/server/services/epm/agent/agent.ts

```diff
@@ -136,6 +136,16 @@ function toJsonHelper(value: any) {
 }
 handlebars.registerHelper('to_json', toJsonHelper);
 
+function percentEncodeHelper(input: string) {
+    let encodedString = encodeURIComponent(input);
+    // handle left out characters
+    encodedString = encodedString
+    .replace(/[!'()*]/g, (char) => '%' + char.charCodeAt(0).toString(16).toUpperCase());
```

Comment:
I see the `%` is coming from here, could we add a comment to explain why it is needed? The percent is not mentioned in the linked [issue](https://github.com/elastic/integrations/issues/5213).

Could we also add manual steps to verify with a test package to the pr description? We usually do this to verify the functionality end to end. If possible, these testcases should be converted to automated API/e2e tests.

Reply:
Added a comment to explain the logic behind using %. Also, updated the PR description to include the steps to test. Finally, added a test case to cover domain\username as mentioned in the referenced issue.